flopy/utils/crs.py

```diff
@@ -0,0 +1,123 @@
+"""Utilities related to coordinate reference system handling.
+"""
+import warnings
+from pathlib import Path
+
+from ..utils import import_optional_dependency
+
+
+def get_authority_crs(crs):
+    """Try to get the authority representation for a
+    coordinate reference system (CRS), for more robust
+    comparison with other CRS objects.
+
+    Parameters
+    ----------
+    crs : pyproj.CRS, optional if `prj` is specified
+        Coordinate reference system (CRS) for the model grid
+        (must be projected; geographic CRS are not supported).
+        The value can be anything accepted by
+        :meth:`pyproj.CRS.from_user_input() <pyproj.crs.CRS.from_user_input>`,
+        such as an authority string (eg "EPSG:26916") or a WKT string.
+
+    Returns
+    -------
+    authority_crs : pyproj.CRS instance
+        CRS instance initiallized with the name
+        and authority code (e.g. epsg: 5070) produced by
+        :meth:`pyproj.crs.CRS.to_authority`
+
+    Notes
+    -----
+    :meth:`pyproj.crs.CRS.to_authority` will return None if a matching
+    authority name and code can't be found. In this case,
+    the input crs instance will be returned.
+
+    References
+    ----------
+    http://pyproj4.github.io/pyproj/stable/api/crs/crs.html
+
+    """
+    pyproj = import_optional_dependency("pyproj")
+    if crs is not None:
+        crs = pyproj.crs.CRS.from_user_input(crs)
+        authority = crs.to_authority()
+        if authority is not None:
+            return pyproj.CRS.from_user_input(authority)
+        return crs
```

Comment:
I'm guessing that `CRS.from_user_input` is called twice to convert (e.g.) `"+proj=tmerc +lat_0=0 +lon_0=-90 +k=0.9996 +x_0=520000 +y_0=-4480000 +datum=NAD83 +units=m +no_defs "` to `EPSG:3070`? This will probably be fine for most situations, but be aware that proj.4 strings don't always relate 1:1 with an authority EPSG code, so there is some guess-work behind the scenes to attempt to match these, using a default minimum confidence of 70% match. More [here](https://pyproj4.github.io/pyproj/stable/gotchas.html#why-does-the-epsg-code-return-when-using-epsg-xxxx-and-not-with-init-epsg-xxxx).

Reply:
Yes- I believe so. This function is actually from some older code that I had written around when pyproj 2 was released. If I remember right there were instances when two CRS that were effectively the same were not getting recognized as such (for example, in trying to decide whether a dataset needed to be reprojected). Looking through the test cases here, it looks like this extra call is redundant for those (for example, the CRS instance produced by the proj string above is evaluated as equal to a CRS instance produced from `3070`). 

But as noted in the link you included, there are other cases where CRS objects are close enough but don't evaluate to equal. This should hopefully catch the subset of those that meet the 70% criterion. The overall goal is make coordinate referencing as easy as possible for users, for example- people who might be trying to load and visualize old models with various proj.4 string references, who might not know much about coordinate reference systems and just want exported information to show up in the right place in a GIS GUI. 

If this ends up being a problem we can always make changes later.